File: tests/fixtures/python/simple.py
"""
A simple Python file for testing the code analyzer and fixer.
"""

import os
import sys
import datetime  # Unused import

def add_numbers(a, b):
    """Add two numbers and return the result."""
    return a + b

def subtract_numbers(a, b):
    """Subtract b from a and return the result."""
    result = a - b  # This local variable is used
    return result

def multiply_numbers(a, b):
    """Multiply two numbers and return the result."""
    product = a * b  # Good: Variable is used
    return product

def divide_numbers(a, b):
    """Divide a by b and return the result."""
    # Issue: Missing check for division by zero
    return a / b

class Calculator:
    """A simple calculator class."""
    
    def __init__(self, initial_value=0):
        """Initialize the calculator with an optional initial value."""
        self.value = initial_value
        self._unused_attr = None  # Unused attribute
    
    def add(self, x):
        """Add x to the current value."""
        self.value += x
        return self
    
    def subtract(self, x):
        """Subtract x from the current value."""
        self.value -= x
        return self
    
    def get_value(self):
        """Return the current value."""
        return self.value
    
    def reset(self):
        """Reset the calculator to zero."""
        old_value = self.value  # Unused variable
        self.value = 0
        return self

# Example usage
if __name__ == "__main__":
    x = 10
    y = 5
    
    # Calculate and print results
    print(f"Add: {add_numbers(x, y)}")
    print(f"Subtract: {subtract_numbers(x, y)}")
    print(f"Multiply: {multiply_numbers(x, y)}")
    
    try:
        print(f"Divide: {divide_numbers(x, y)}")
    except ZeroDivisionError:
        print("Cannot divide by zero")
    
    # Create and use calculator
    calc = Calculator(x)
    calc.add(y).subtract(3)
    print(f"Calculator: {calc.get_value()}")

File: tests/fixtures/csharp/simple.cs
using System;
using System.Collections.Generic;
using System.Linq;  // Unused namespace

namespace CodeRefactorTests
{
    /// <summary>
    /// A simple calculator class for testing the code analyzer and fixer.
    /// </summary>
    public class Calculator
    {
        // Unused field
        private int unusedField;

        /// <summary>
        /// Adds two numbers.
        /// </summary>
        /// <param name="a">First number</param>
        /// <param name="b">Second number</param>
        /// <returns>The sum of a and b</returns>
        public int Add(int a, int b)
        {
            return a + b;
        }

        /// <summary>
        /// Subtracts the second number from the first.
        /// </summary>
        /// <param name="a">First number</param>
        /// <param name="b">Second number</param>
        /// <returns>The result of a - b</returns>
        public int Subtract(int a, int b)
        {
            return a - b;
        }

        /// <summary>
        /// Multiplies two numbers.
        /// </summary>
        /// <param name="a">First number</param>
        /// <param name="b">Second number</param>
        /// <returns>The product of a and b</returns>
        public int Multiply(int a, int b)
        {
            return a * b;
        }

        /// <summary>
        /// Divides the first number by the second.
        /// </summary>
        /// <param name="a">First number</param>
        /// <param name="b">Second number</param>
        /// <returns>The result of a / b</returns>
        public double Divide(int a, int b)
        {
            // Missing null check
            return a / b;  // Potential division by zero
        }

        /// <summary>
        /// Calculates the average of a list of numbers.
        /// </summary>
        /// <param name="numbers">List of numbers</param>
        /// <returns>The average of the numbers</returns>
        public double Average(List<int> numbers)
        {
            // Issue: Potential null reference exception
            int sum = 0;
            foreach (int number in numbers)
            {
                sum += number;
            }
            
            return sum / numbers.Count;  // Potential division by zero
        }
    }

    /// <summary>
    /// Main program class.
    /// </summary>
    public class Program
    {
        /// <summary>
        /// Entry point of the application.
        /// </summary>
        /// <param name="args">Command line arguments</param>
        public static void Main(string[] args)
        {
            // Unused variable
            string Greeting = "Hello, World!";
            
            // Create a calculator
            Calculator calculator = new Calculator();
            
            // Perform calculations
            int a = 10;
            int b = 5;
            
            Console.WriteLine($"Add: {calculator.Add(a, b)}");
            Console.WriteLine($"Subtract: {calculator.Subtract(a, b)}");
            Console.WriteLine($"Multiply: {calculator.Multiply(a, b)}");
            
            // Issue: No try-catch for potential exception
            Console.WriteLine($"Divide: {calculator.Divide(a, b)}");
            
            // Create a list of numbers
            List<int> numbers = new List<int> { 1, 2, 3, 4, 5 };
            
            // Calculate and display the average
            double average = calculator.Average(numbers);
            Console.WriteLine($"Average: {average}");
            
            // Wait for user input before exiting
            Console.WriteLine("Press any key to exit...");
            Console.ReadKey();
        }
    }
}